raiden/api/python.py

```diff
@@ -276,7 +289,7 @@ def token_swap_and_wait(
             taker_amount,
             taker_address,
         )
-        async_result.wait()
+        return async_result.get(timeout=timeout)
```

Comment:
Don't do this, there are two methods for a reason, if the caller wants to control the wait timeout it must use the `token_swap_async` and call `wait(timeout)`.

Think about it, adding a timeout as an argument through multiple layers is a recipe for bad code, every new function needs to set a default value, which may turn out different because of no good reason, and every layer must pass downwards it's notion of timeout, being quite easy to forget while refactoring.

returning an async result allows the caller to do whatever it pleases, it may wait forever or use a timeout, there is no loss of generality in the code, and we don't need to worry about the above.

Reply:
Ok, just wanted to make it consistent with the `transfer_and_wait` call, which *already* takes an optional `timeout` parameter. A lot of those APIs are called _aliased_ or mocked by the REST API, which leads to REST endpoints dependent on `async_result` resolution, and may hang forever. We shouldn't allow that on the REST interface, and a `timeout` is preferably to requests which never returns. But, I agree, it's better to set this kind of behavior on the caller's.